src/ucp/core/ucp_mm.c

```diff
@@ -126,6 +127,16 @@ ucs_status_t ucp_mem_rereg_mds(ucp_context_h context, ucp_md_map_t reg_md_map,
                 ucp_memory_detect_internal(context, address, length, &mem_info);
                 base_address = mem_info.base_address;
```

Comment:
i'd move `base_address` and `reg_length` initialization from lines 123-124 to the `else` branch of this `if`